### PR TITLE
Document supported polars methods

### LINKED ISSUE #2025: Polars: documentation page with tables of supported APIs
Move this information, cut down and repurpose:
https://docs.google.com/spreadsheets/d/1w4H-qrSe9pkvNT2NhowfiIHNWXSseOyMX2nu5H6-RkQ/edit?gid=0#gid=0

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #2025

Draft for now:
- Is this the right place for this information? I think at one time there was an idea of adding a page in the tabular data section, but this seems more like reference material.
- Is a fourth column with doctests useful? Maybe they are useful, but cramming them in a table is a mistake.
- How should we explain where these can be used? They can't be used right after the `dp`, but they will work if `noise` or another method intervenes.
- Does it make sense to do the same thing for `LazyFrame` methods?

### CODE REVIEW

File: python/src/opendp/extras/polars/__init__.py [hunk 1]
```diff
@@ -44,6 +44,53 @@ class DPExpr(object):
...
+    In addition to the DP-specific methods documented below, many Polars ``Expr`` methods are also supported.
+    For these, the best documentation is the official Polars documentation.
+
+    .. list-table:: Supported Polars ``Expr`` Methods
+        :header-rows: 1
+
+        * - Method
+          - Operators
+          - Use for
+          - Usage
```
Comment (Shoeboxam) [inline python/src/opendp/extras/polars/__init__.py:57]: To save space, how would you feel about condensing into two columns:

* method (with link)
* notes

Where for the most part, notes talks about privacy-specific behavior, but if space allows, also some pointers. 

Some suggestions:

# Context Transformations

.
* with_columns
* expressions must not reorder or add/remove rows

.
* filter
* margin descriptors may be lost; consider group_by instead

# Context Aggregations

.
* select
* can specify multiple expressions

.
* group_by/agg
* can specify multiple grouping keys and expressions



# Expression Transformations
.
* eq, ne, lt, le, gt, ge
* Or use operators: ``==`` ``!=`` ``<`` ``<=`` ``>`` ``>=``

.
* clip
* establishes bounds descriptor

.
* fill_null
* if fill value is non-null, establishes non-nullity descriptor

.
* fill_nan
* if fill value is non-null and non-nan, establishes non-nan descriptor

# Expression Aggregators
.
* len
* number of rows, including nulls

.
* count
* number of rows, not including nulls

.
* n_unique
* number of unique rows

I think code examples are already covered by Polars. Please don't hesitate to disagree if you feel differently!

**Review by Shoeboxam — DISMISSED** [review]
Just sharing thoughts, I'm flexible.

**Comment by mccalluc** [comment]
Good call.
- I've broken this into two separate tables, because they work differently, though this doc doesn't really explain the difference.
- I'm not covering the few methods you list above for which we have an actual implementation: Including them in the table seems redundant.
- For me the code example would really have been a test for us. There's not room in a table to give an example that's useful to users.

**Comment by mccalluc** [comment]
Tons of CI failures; Re-running now: Probably a github problem?

**Comment by mccalluc** [comment]
Re-run also failed? re-running again...

File: python/src/opendp/extras/polars/__init__.py (hunk 1 above)
Comment (mccalluc) [reply python/src/opendp/extras/polars/__init__.py]: Thanks. Should be addressed.

File: python/src/opendp/extras/polars/__init__.py [hunk 2]
```diff
@@ -44,6 +44,43 @@ class DPExpr(object):
...
+    .. list-table:: Supported Polars ``Expr`` Methods
+        :header-rows: 1
+
+        * - Method
+          - Comments
+        * - `alias <https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.alias.html>`_
+          - Rename the expression
+        * - `eq, ne, lt, le, gt, ge <https://docs.pola.rs/api/python/stable/reference/expressions/operators.html#comparison>`_
+          - Comparison operators may be more readable: ``==`` ``!=`` ``<`` ``<=`` ``>`` ``>=``
+        * - `and, or, xor <https://docs.pola.rs/api/python/stable/reference/expressions/operators.html#conjunction>`_
```
Comment (Shoeboxam) [inline python/src/opendp/extras/polars/__init__.py:60]:
```suggestion
        * - `and_, or_, xor <https://docs.pola.rs/api/python/stable/reference/expressions/operators.html#conjunction>`_
```

File: python/src/opendp/extras/polars/__init__.py [hunk 3]
```diff
@@ -44,6 +44,43 @@ class DPExpr(object):
...
+        :header-rows: 1
+
+        * - Method
+          - Comments
+        * - `len <https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.len.html>`_
+          - Number of rows, including nulls
+        * - `count <https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.count.html>`_
+          - Number of rows, not including nulls
+        * - `n_unique <https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.n_unique.html>`_
+          - Number of unique rows
```
Comment (Shoeboxam) [inline python/src/opendp/extras/polars/__init__.py:83]: I had previously written this to give examples, but we don't have n_unique. Instead, add `sum`?

**Review by Shoeboxam — CHANGES_REQUESTED** [review]
Some small changes. Looking good.

File: python/src/opendp/extras/polars/__init__.py (hunk 3 above)
Comment (Shoeboxam) [reply python/src/opendp/extras/polars/__init__.py:83]: Bump on this ^^

**Review by Shoeboxam — CHANGES_REQUESTED** [review]
I guess my previous comment didn't go in as a review? My bad.

**Comment by mccalluc** [comment]
Sorry to miss that. Fixing now...

**Review by Shoeboxam — APPROVED** [review]
Thanks!

**Pull request merged**